backend/src/main/java/com/huellapositiva/application/controller/ESALApiController.java

```diff
@@ -99,6 +104,52 @@ public void registerESAL(@Validated @RequestBody ESALRequestDto dto,
         }
     }
 
+    @Operation(
+            summary = "Updates the information of an ESAL",
+            description = "Updates the information of the ESAL linked to the logged employee. Roles allowed CONTACT_PERSON and CONTACT_PERSON_NOT_CONFIRMED.",
+            tags = "ESAL",
+            parameters = {
+                    @Parameter(name = "X-XSRF-TOKEN", in = ParameterIn.HEADER, required = true, example = "a6f5086d-af6b-464f-988b-7a604e46062b", description = "For take this value, open your inspector code on your browser, and take the value of the cookie with the name 'XSRF-TOKEN'. Example: a6f5086d-af6b-464f-988b-7a604e46062b"),
+                    @Parameter(name = "XSRF-TOKEN", in = ParameterIn.COOKIE,required = true, example = "a6f5086d-af6b-464f-988b-7a604e46062b", description = "Same value of X-XSRF-TOKEN")
+            },
+            security = {
+                    @SecurityRequirement(name = "accessToken")
+            }
+    )
+    @ApiResponses(
+            value = {
+                    @ApiResponse(
+                            responseCode = "200",
+                            description = "Ok, ESAL updated successfully."
+                    ),
+                    @ApiResponse(
+                            responseCode = "400",
+                            description = "Bad Request, unable to validate provided data."
+                    ),
+                    @ApiResponse(
+                            responseCode = "409",
+                            description = "Conflict, the provided name is already taken."
+                    ),
+                    @ApiResponse(
+                            responseCode = "500",
+                            description = "Internal server error, could not update the esal.",
+                            content = @Content(mediaType = "application/json")
+                    )
+            }
+    )
+    @PostMapping("/update")
+    @RolesAllowed({"CONTACT_PERSON", "CONTACT_PERSON_NOT_CONFIRMED"})
+    public void updateESAL(@Validated @RequestBody UpdateESALDto dto,
+                           @Parameter(hidden = true) @AuthenticationPrincipal String accountId){
+        try{
+            updateESALAction.execute(dto, accountId);
+        } catch (DataIntegrityViolationException ex){
+            throw new ResponseStatusException(HttpStatus.CONFLICT, "ESAL named " + dto.getName() + " already exists.");
+        } catch (UserNotFoundException ex){
+            throw new ResponseStatusException(HttpStatus.INTERNAL_SERVER_ERROR, "Could not find the user due to a connectivity issue.");
```

Comment:
UserNotFound and internal server error are two completely different things.

Not found is not a server error, it is a client error indeed which occurred due to a client request something that did not exist

Think which of the two fits your use case

Reply:
Currently, we have changed it to throw a HttpStatus.NOT_FOUND for consistency, but we are still unsure about this. This exception is thrown when the system fetches the current user's data. It's not that the user has requested a non-existent user, but that the system cannot get that user's data, although they are logged in. That's why we thought it should throw a server error instead of a client one.